aws/ami_test.go

```diff
@@ -7,11 +7,31 @@ import (
 	awsgo "github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/cloud-nuke/logging"
 	"github.com/gruntwork-io/cloud-nuke/util"
 	"github.com/gruntwork-io/gruntwork-cli/errors"
 	"github.com/stretchr/testify/assert"
 )
 
+func waitUntilImageAvailable(svc *ec2.EC2, input *ec2.DescribeImagesInput) error {
+	for i := 0; i < 50; i++ {
+		output, err := svc.DescribeImages(input)
+		if err != nil {
+			return err
+		}
+
+		if *output.Images[0].State == "available" {
+			return nil
+		}
+
+		time.Sleep(5 * time.Second)
+		logging.Logger.Debug("Waiting for ELB to be available")
+	}
+
+	// return ElbDeleteError{}
+	return nil
```

Comment:
Wait, shouldn't this return an error, as you weren't able to find the image you wanted even after 50 retries?

Reply:
This was a complete oversight. Thanks for the call out